### PR TITLE
Disable RIS export option when imported dataset is not RIS

### PR DESCRIPTION
**Pull request by terrymyc**
This PR disabled the option to export the RIS file when the imported dataset is not a RIS file. This is controlled dynamically through the REST API.

image

### CODE REVIEW

**Comment by JTeijema** [comment]
Looks like the perfect solution :)

File: asreview/webapp/src/ProjectComponents/ExportComponents/ExportPage.js [hunk 1]
```diff
@@ -91,6 +93,10 @@ const ExportPage = (props) => {
     setExporting(true);
   };
 
+  const disableRIS = () => {
+    return props.dataReader !== "ris-reader";
+  };
+
```
Comment (J535D165) [inline asreview/webapp/src/ProjectComponents/ExportComponents/ExportPage.js:99]: This should be moved to the class/backend.

File: asreview/webapp/src/ProjectComponents/ExportComponents/ExportPage.js [hunk 2]
```diff
@@ -91,6 +93,10 @@ const ExportPage = (props) => {
     setExporting(true);
   };
 
+  const disableRIS = () => {
+    return !props.dataWriter.includes("ris-writer");
+  };
```
Comment (terrymyc) [inline asreview/webapp/src/ProjectComponents/ExportComponents/ExportPage.js:98]: @J535D165 Available writers are returned in the API, but we still need an indicator to disable an option.

**Comment by ottomattas** [comment]
The screenshot will be added to the documentation with the following PRs: https://github.com/asreview/asreview/pull/990, https://github.com/asreview/asreview/pull/966.

**Comment by terrymyc** [comment]
> The screenshot will be added to the documentation with the following PRs: #990, #966.

@ottomattas Which screenshot do you use? The one in this PR does not meet our [screenshot requirement](https://github.com/asreview/asreview/blob/master/DEVELOPMENT.md#screenshots)

**Comment by ottomattas** [comment]
@terrymyc I used the screenshot you posted in your first comment in the current PR. Would you be able to share a screenshot that follows the requirements?

**Comment by terrymyc** [comment]
@ottomattas Sorry for the late response. The RIS option will not be visible in the frontend if the imported dataset is not a RIS file. So no more screenshots for this PR...

File: asreview/io/csv_writer.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    fp: str, NoneType
+        Filepath or None for buffer.
+    sep: str
+        Seperator of the file.
+    labels: list, numpy.ndarray
+        Current labels will be overwritten by these labels
+        (including unlabelled). No effect if labels is None.
+    ranking: list
+        Reorder the dataframe according to these (internal) indices.
+        Default ordering if ranking is None.
```
Comment (J535D165) [inline asreview/io/csv_writer.py:31]: Actually, this should be in the `write_data` method. The arguments here should describe the `__init__`. I think there are none.

File: asreview/data/base.py [hunk 4]
```diff
@@ -419,71 +455,3 @@ def to_dataframe(self, labels=None, ranking=None):
             result_df.loc[result_df[col_label] == LABEL_NA, col_label] = np.nan
 
         return result_df
-
-    def to_csv(self, fp, sep=",", labels=None, ranking=None):
-        """Export to csv.
-
```
Comment (J535D165) [inline asreview/data/base.py:425]: Makes sense.

File: asreview/data/base.py [hunk 5]
```diff
@@ -362,14 +396,16 @@ def to_file(self, fp, labels=None, ranking=None):
         ranking: list, numpy.ndarray
             Optionally, dataframe rows can be reordered.
         """
+        df = self.to_dataframe(labels=labels, ranking=ranking)
+
         if Path(fp).suffix in [".csv", ".CSV"]:
```
Comment (J535D165) [inline asreview/data/base.py:401]: This should be based on the available entry points isn't it?

Comment (J535D165) [reply asreview/data/base.py:401]: Like in from_file

**Review by J535D165 — CHANGES_REQUESTED** [review]
Great work, some suggestions, and questions. Maybe more will follow.

File: asreview/data/base.py (hunk 5 above)
Comment (terrymyc) [reply asreview/data/base.py:401]: Should `export_to_string` function in `project.py` be integrated into `to_file`?

**Review by J535D165 — APPROVED** [review]

**Comment by J535D165** [comment]
Great work and what a journey :)

**Pull request merged**